### PR TITLE
Source Jira: Small fix in the board stream

### PR DESCRIPTION
**Pull request by anamargaridarl**
## What
Some boards don't have projects associated with them. Thus, some records don't have the fields inside the location dict. I created a fail-safe approach. 

## How
Use get instead. The project fields will be None if no project is available.

## Recommended reading order
1. streams.py

## 🚨 User Impact 🚨
Boards stream won't break

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

Passing

</details>

<details><summary><strong>Integration</strong></summary>

Passing
</details>

<details><summary><strong>Acceptance</strong></summary>

Passing

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 1]
```diff
@@ -190,12 +190,16 @@ def path(self, **kwargs) -> str:
 
     def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
         for board in super().read_records(**kwargs):
-            if not self._projects or board["location"]["projectKey"] in self._projects:
+            location = board["location"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:193]:
```suggestion
            location = board.get("location", {})
```
This at least won't break the code in any case.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments. I started the tests to check if it will work.

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 2]
```diff
@@ -190,12 +190,16 @@ def path(self, **kwargs) -> str:
...
                 yield board
 
     def transform(self, record: MutableMapping[str, Any], stream_slice: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
-        record["projectId"] = str(record["location"]["projectId"])
-        record["projectKey"] = record["location"]["projectKey"]
+        if record.get("location"):
+            location = record.get("location")
+            if location:
+                record["projectId"] = str(location.get("projectId"))
+                record["projectKey"] = location.get("projectKey")
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:202]:
```suggestion
        location = record.get("location")
        if location:
            record["projectId"] = str(location.get("projectId"))
            record["projectKey"] = location.get("projectKey")
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
forget one suggestion

**Comment by anamargaridarl** [comment]
Thank you for the review! I applied the changes requested @marcosmarxm

**Comment by Logicoeur** [comment]
Bump. encountered this error too

**Review by natalyjazzviolin — APPROVED** [review]
Looks good!

**Review by marcosmarxm — APPROVED** [review]

**Comment by quorak** [comment]
Awesome fix. When can we expect a release?

**Pull request merged**